Maid 1459 hole punch example

inetic opened the pull request:
<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/maidsafe/crust/371)

<!-- Reviewable:end -->

inetic committed:
Adds readline support to crust_peer example

vinipsmaker committed:
rendezvous uTP connection API

Necessary to use the hole punched UDP sockets.

Missing:

- More documentation.
- Tests.

canndrew committed:
Replace WrapSocketAddr with SocketAddrW

inetic committed:
Updates command documentation in crust_peer and add 'list' command

inetic committed:
Adds commands to crust_peer to do rendezvous uTP connect

maidsafe-highfive commented:
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

inetic committed:
Removes support for readline

Windows CI machines did not have the lib

vinipsmaker committed:
add doc and test for `service::rendezvous_connect`

vinipsmaker committed:
workaround to let CI tests pass until issue 374 is solved

inetic committed:
In crust_peer example, handle OnRendezvousConnect

inetic committed:
crust_peer now prints command usage right after started

Fraser999 committed:
Merge branch 'MAID-1459-hole-punch-example' of https://github.com/inetic/crust into MAID-1459-hole-punch-example

Fraser999 committed:
Added "send-all" option to crust_peer example.

Commit:
Merge pull request #1 from Fraser999/MAID-1459-hole-punch-example

Added "send-all" option to crust_peer example.

inetic committed:
Info output for FileHandler

vinipsmaker committed:
Revert "workaround to let CI tests pass until issue 374 is solved"

This reverts commit a8256a0b0e38122f923681ea448279ac91cbafa6.

vinipsmaker committed:
Merge branch 'master' into MAID-1431

inetic committed:
Merge remote-tracking branch 'vin/MAID-1431' into MAID-1459-hole-punch-example

inetic committed:
Merge branch 'master' into MAID-1459-hole-punch-example

Conflicts:
	src/state.rs

inetic committed:
Revert "Info output for FileHandler"

This reverts commit 56c9be71bb35418064b2db46652e20511a5a4be2.

inetic committed:
Temporarily ignore the rendezvous_connection test

Should be fixed once PR to rust-utp
https://github.com/maidsafe/rust-utp/pull/5
gets merged

vinipsmaker commented:
@inetic, I release a new `temp_utp` version with your changes and now tests pass correctly: https://github.com/maidsafe/crust/pull/366 (please don't pull the dummy commit, it was done only to trigger the CI that somebody has cancelled)

I think you can revert/discard commit 7b82dbd9a796eb9b597fd2bceb49597727368566 already.
